mantidproject/mantid
# Added 'alwaysHidden' tag to infinite components in IDF

**Description of work.**

Following on from the changes in #19683, a flag `HiddenInInstrumentView` to allow the user to specify components which should not be rendered. If this flag is detected, the component is not rendered. The bounding-box check is left in place, but displays a warning to the user that they tried to render a component which was too big.

**To test:**

Run `LoadEmptyInstrument` on the updated `WISH_Definition_10Panels.xml`. In full 3D view, the scale should now be set correctly. Now run with the IDF where the infinite component is not explicitly hidden
([WISH_Definition_10Panels.txt](https://github.com/mantidproject/mantid/files/1172529/WISH_Definition_10Panels.txt)) - the scale should still be set correctly, and a warning shown that a component was too big.



Fixes #19684 #19539 . 

**Release Notes** 

[here](https://github.com/mantidproject/mantid/pull/20084/commits/ae4fbca399ba8b15a9c5e380c2ef8a00d0398654)


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
